saleor/graphql/account/mutations/base.py

```diff
@@ -414,6 +354,34 @@ def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
             if user_gift_cards := get_user_gift_cards(instance):
                 mark_gift_cards_search_index_as_dirty(user_gift_cards)
 
+    @classmethod
+    def save_default_addresses(
+        cls, *, cleaned_input: dict, user_instance: models.User, save_user: bool
+    ):
+        default_shipping_address: models.Address | None = cleaned_input.get(
+            SHIPPING_ADDRESS_FIELD
+        )
+
+        if default_shipping_address:
+            default_shipping_address.save()
+            user_instance.default_shipping_address = default_shipping_address
+
+        default_billing_address: models.Address | None = cleaned_input.get(
+            BILLING_ADDRESS_FIELD
+        )
+
+        if default_billing_address:
+            default_billing_address.save()
+            user_instance.default_billing_address = default_billing_address
+
+        if save_user:
+            user_instance.save()
```

Comment:
I think this save shouldn't be here. I was looking for save in `CustomerCreate` and I won't expect that the user will be saved in `save_default_addresses` method.

Reply:
I think that this is for handling the case, when you need to save addresses on the instance without the `pk`. 
As addresses are palced in `default_...address` fields, and as a foreign relation: `addresses`. You need to have user represantation on the database side, to perfomr proper address save